### ISSUE TITLE
Handling custom licenses through Zenodo API

### ISSUE THREAD

**Issue by slugb0t** [zenodo/zenodo#2547]
Hi everyone,

I've encountered an issue with Zenodo's API in handling custom licenses.

When saving a draft with a custom license through Zenodo's UI, the network tab shows that the custom license is saved with specific keys and values in the rights key. It also seems assigns a default **en** (language) key, even though I haven’t specified one in the metadata.

**However, when I attempt to save this same custom license via the API, it doesn’t persist.** Despite setting the identical keys and values as seen in the UI, the deposition defaults to a CC-BY-4.0 license instead.

Sandbox UI Example:
![image](https://github.com/user-attachments/assets/a2510794-81a4-40b1-8d7a-6e4fb8bc191f)

API Example (through postman):
![image](https://github.com/user-attachments/assets/e47b0966-5773-4845-888c-18cc1fcf69a4)

How do I adjust the API request to ensure the custom license is saved correctly? Let me know if there’s anything I can clarify further or if additional details would help.

**Comment by cjw85** [comment]
@slugb0t Did you ever figure this out? I'm experiencing the same.

**Comment by slugb0t** [comment]
@cjw85 Nope! Was hoping I'd hear back from the Zenodo team. Sadly their API documentation has no mention on custom licenses but their web app does allow for this.
If you happen to figure this out let me know please!

**Comment by cjw85** [comment]
I actually did manage to do this just two days ago after coming back to the problem. I had no luck following the documented APIs so I just ignored them and just poked around in chrome console to observe the requests being made. I've attached the Python client class I drafted for performing requests.

The class has a hard-coded `OWNER` attribute, and within the `.update_meta()` method there is some hard-coded data for the license I wanted to attach. The process is reasonably straight-forward, but I found edge cases where the draft template the API sends you is not valid for pushing back to them. 

[client.py](https://github.com/user-attachments/files/22278027/client.py)

**Comment by slugb0t** [comment]
@cjw85 Thanks for sharing! What I was missing then was a url to the custom license. I'll test this out myself soon
